### PR TITLE
[Uid] Add a caution message about using UUIDs as primary keys

### LINKED ISSUE #19705: [Uid] Add a note on the use of entity primary keys
It would be interesting to add a note on the use U*id values for primary key in an entity. 
This use is bad practice.
What do you think ?

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #19705.

### CODE REVIEW

**Comment by seb-jean** [comment]
And for ULIDs ?

**Comment by javiereguiluz** [comment]
@seb-jean I updated the PR with another caution about ULIDs. Thanks.

File: components/uid.rst [hunk 1]
```diff
@@ -344,6 +344,14 @@ entity primary keys::
         // ...
     }
 
+.. caution::
+
+    Using UUIDs as primary keys is usually not recommended for performance reasons:
+    indexes are slower and take more space (because UUIDs in binary format take 128 bits
+    instead of 32/64 bits for auto-incremental integers) and the non-sequential nature of
+    UUIDs fragments indexes. UUID v7 is the only variant that solves the fragmentation
```
Comment (alamirault) [inline components/uid.rst:352]: WDYT to add anchor link on UUUD v7 ?

Comment (javiereguiluz) [reply components/uid.rst:352]: Good idea! I did this while merging. Thanks.

Comment (nicolas-grekas) [reply components/uid.rst:352]: UUIDv6 are also fine from a fragmentation PoV

Comment (javiereguiluz) [reply components/uid.rst:352]: I've updated the docs here: https://github.com/symfony/symfony-docs/commit/2cbc55e95efc57d0204684843732ddfee3507f06 Thanks!

**Pull request merged**